docs: add plan for node rejected details and more (#12564)
hashicorp/nomad: website/content/docs/operations/monitoring-nomad.mdx

for a server to apply Raft entries to the internal state machine. If
this number trends upwards, look at the `nomad.nomad.fsm.*` metrics to
see if a specific Raft entry is increasing in latency. You can compare
this to warn-level logs on the Nomad servers for `attempting to apply
large raft entry`. If a specific type of message appears here, there
may be a job with a large job specification or dispatch payload that
is increasing the time it takes to apply raft messages.

## Scheduling

The [Scheduling] documentation describes the workflow of how evaluations become
scheduled plans and placed allocations.

### Progress

There is a class of bug possible in Nomad where the two parts of the scheduling
pipeline, the workers and the leader's plan applier, *disagree* about the
validity of a plan. In the pathological case this can cause a job to never

> finish scheduling as workers produce the same plan the plan applier repeatedly
> rejects.

Reviewer: This might make this sentence flow a bit better?

```suggestion
finish scheduling, as workers produce the same plan and the plan applier repeatedly
rejects it.
```

> While this class of bug if very rare, it can be detected by repeated log lines
> on the Nomad servers containing `plan for node rejcted`:

Reviewer: ```suggestion
While this class of bug is very rare, it can be detected by repeated log lines
on the Nomad servers containing `plan for node rejected`:
```

```
nomad: plan for node rejected: node_id=0fa84370-c713-b914-d329-f6485951cddc reason="reserved port collision" eval_id=098a5
```

While it is possible for these log lines to occur infrequently due to normal
cluster conditions, they should not appear repeatedly and prevent the job from
eventually running (look up the evaluation ID logged to find the job).

> If this log *does* appear repeatedly with the same `node_id` referenced. Trying

Reviewer: ```suggestion
If this log *does* appear repeatedly with the same `node_id` referenced, try
```
